network/802.11/eapol_key.ksy

```diff
@@ -0,0 +1,8 @@
+meta:
+  id: eapol_key
+  title: IEEE 802.11i EAPOL
+  license: Unlicense
+doc-ref: IEEE 802.11
+seq:
+  - id: key
+    size: 256
```

Comment:
Generally, I'm not too keen on adding such "placeholder" types for types. For example, in many applications, keys and nonces are literally just arrays of (random) bytes of a certain length. They'll never be more than that, and I heavily doubt that there would be any better replacements for them in any languages (expect for may be trivial containers). Right now all they do is add *lots* of unnecessary complication for both ksy writers and end-users.

Besides, in many cases, such as "nonce", "mac_address", or even "eapol_key" is a way too ambiguous name. For example, there are several items that can be referred to as "EAPOL keys":

* EAPOL-Key Confirmation Key (KCK) - 16 bytes
* EAPOL-Key Encryption Key (KEK) - 16 bytes
* whole EAPOL-Key frame

And, yeah, I still have to remind you of EOF newlines.

Reply:
I don't want to remember eapol key length everywhere where it can encounter, so I moved it into a separate type. It's easier to maintain. Also it gives self-documentation. Looking at the type name we understand what is it.

About eol in the eof. Why do we need these eol's in the end of files? They are redundant, since the file is ended. They may pay somewhere, for example if you copy the file content into a memory buffer anf replace the ```\n```s with nullbytes, but a good program cannot rely on that. So I vote for removing empty lines from the end of all the ksy files in the repo. See the proposed .editorconfig PR.

Reply:
As I've noted above, there's basically no such thing as "eapol key length". Even in the context of "eapol" buffer in hccap, it's variable length field that takes up to 256 bytes, padded to 256 bytes to make up a fixed-size C struct — as described [in their wiki](https://hashcat.net/wiki/doku.php?id=hccapx#detailed_structure). So, this "eapol_key.ksy" is actually wrong even in respect to hccap, and I highly doubt that anyone else would use exactly same 256-byte buffer.

> It's easier to maintain.

I really doubt that. Writing `size: XXX` vs adding separate dozen-line file.

> Also it gives self-documentation. Looking at the type name we understand what is it.

In majority of the cases, you'll have an attribute named in a way that it gives out its meaning. And, on the other hand, it's very easy to fall into trap of extracting a certain type as some sort of "type of global interest", while it turns out that it's a minor internal thing. You've already felt into it with "eapol_key.ksy". "mac_address.ksy" is probably also a bad idea, but due to another reason: it probably should be platform-specific external type, and having `imports` of some KS type makes that more complicated.

> So I vote for removing empty lines from the end of all the ksy files in the repo.

This is definitely not a place to discuss it, and it's definitely not a good idea to have some files in one way and some in the other. Please state your arguments in https://github.com/kaitai-io/kaitai_struct/issues/140, but please do some research and come with the hard facts, not just opinions.